### PR TITLE
Add default config set for Optimism

### PR DESCRIPTION
**Pull request by samsondav**

### CODE REVIEW

File: core/store/orm/config.go [hunk 1]
```diff
@@ -158,9 +158,27 @@ func init() {
...
 
+	// Optimism is an L2 chain. Pending proper L2 support, for now we rely on Optimism's sequencer
+	optimism := ChainSpecificDefaultSet{
+		EthGasBumpThreshold:              0, // Never bump gas on optimism
+		EthFinalityDepth:                 1, // Sequencer offers absolute finality as long as no re-org longer than 20 blocks occurs on main chain, this event would require special handling (new txm)
+		EthHeadTrackerHistoryDepth:       10,
+		EthBalanceMonitorBlockDelay:      0,
+		EthTxResendAfterThreshold:        5 * time.Second,
+		GasUpdaterBlockHistorySize:       0,               // Force an error if someone set GAS_UPDATER_ENABLED=true by accident; we never want to run the gas updater on optimism
+		HeadTimeBudget:                   1 * time.Second, // Actually heads on Optimism happen every time a transaction is sent so it could be much more frequent than this. Will need to observe in practice how rapid they are and maybe implement special casing
```
Comment (krebernisak) [inline core/store/orm/config.go:169]: Should we immediately lower this to 1/60 s as we expect ~60 tps?

Comment (samsondav) [reply core/store/orm/config.go:169]: The node can't handle that.

Comment (krebernisak) [reply core/store/orm/config.go:169]: Would it make sense to start a little higher, like `0.1 s` for example. That is 1/6 of the expected capacity.

**Review by krebernisak — DISMISSED** [review]
Looks good!

Thanks, @samsondav! Let's test it out 🚀

**Review by krebernisak — APPROVED** [review]

**Pull request merged**